### DISCUSSION TITLE
［争议］是否应当为同名lib做多套方案来切换

### DISCUSSION THREAD

**Discussion by binx6** [LibChecker/LibChecker#908] [General]
**随着越来越多的规则被贡献（这里特指`原生库`方面），我意识到一个问题，即有些`so`文件并非是单一存在情况的，还会出现同名可能，比如我上次反馈的~~cJSON~~的问题，确实是cJSON编译之后产生，但是在AndroLua_pro项目（成品为AndroLua+，作者nirenr）中所生成的`so`，却并非来自cJSON，而是[这样的](https://github.com/nirenr/AndroLua_pro/tree/master/app/src/main/jni/cjson)，两者生成so却为同名，不仅如此，其它某些so文件也存在这个问题。考虑到此因素，所以我提议是否应当为同名lib做多套方案来切换**

**Comment by binx6** [comment]
@zhaobozhen @qhy040404

**Comment by qhy040404** [comment]
那如何来做识别呢

**Comment by binx6** [comment]
> 那如何来做识别呢

我目前有三个想法，但是实际应用起来感觉不太行
①：dump操作，拿到伪代码进行特征识别
②：字节/大小判断，选一个区间……
③：在卡片里直接切换（加个说明`该so可能来源于`什么的）

**Reply by zhaobozhen** [reply]
前两个方法太复杂了，不同的原声库有不同的判断方法，实际上现在已经有类似的方法在判断一小部份原声库了。比较轻量的方案是方案三，可能在未来版本加入。

**Reply by su1216** [reply]
支持方案3

**Reply by binx6** [reply]
> 支持方案3

我个人也偏向方案三
